onchain/danaswap/DanaSwap.hs

```diff
@@ -47,6 +75,50 @@ instance DerivePlutusType LiquidityRedeemer where type DPTStrat _ = PlutusTypeNe
 instance PTryFrom PData (PAsData LiquidityRedeemer)
 instance PShow LiquidityRedeemer
 
+newtype ConfigData (s :: S)
+  = ConfigData
+      ( Term
+          s
+          ( PDataRecord
+              '[ "poolAdr" ':= PAddress
+               , "liquidityCS" ':= PCurrencySymbol
+               ]
+          )
+      )
+  deriving stock (Generic)
+  deriving anyclass (PlutusType, PIsData, PEq)
+
+instance DerivePlutusType ConfigData where type DPTStrat _ = PlutusTypeNewtype
+instance PTryFrom PData (PAsData ConfigData)
+
+type PAssetClass = PBuiltinPair (PAsData PCurrencySymbol) (PAsData PTokenName)
+
+newtype PoolData (s :: S)
+  = PoolData
+      ( Term
+          s
+          ( PDataRecord
+              '[ "ac1" ':= PAssetClass
+               , "ac2" ':= PAssetClass
+               , "bal1" ':= PInteger
+               , "bal2" ':= PInteger
+               , "adminBal1" ':= PInteger
+               , "adminBal2" ':= PInteger
+               , "issuedLiquidity" ':= PInteger
+               , "isLive" ':= PBool
```

Comment:
Could we keep the names of the properties in this datum the same as the ones we're sending from the front-end?

Reply:
sure, I like these names better so I made the off-chain agree with this.